src/oic/oauth2/util.py

```diff
@@ -55,9 +59,15 @@ def get_or_post(uri, method, req, content_type=DEFAULT_POST_CONTENT_TYPE,
     :return:
     """
     if method in ["GET", "DELETE"]:
-        _qp = req.to_urlencoded()
-        if _qp:
-            path = uri + '?' + _qp
+        if req.keys():
+            comp = urlsplit(uri)
+            if comp.query:
+                req = req.copy()
```

Comment:
Please make this a first statement in the function to make it clear and hopefully error prone :)